Editor/Scripts/BotManagement/RGBotSynchronizer.cs

```diff
@@ -230,61 +244,94 @@ private async Task<List<RGBot>> GetBotsFromRG()
         {
-            var localMD5 = new Dictionary<long, string>();
+            // Fetch code details from the server.
+            RGBotCodeDetails botCodeDetails = null;
+            await _rgServiceManager.GetBotCodeDetails(
+                remoteBot.id,
+                (details) =>
+                {
+                    botCodeDetails = details;
+                },
+                () => { }
+            );
 
-            // create zip files for all the bots I have locally
-            foreach (var rgBot in localBots)
+            if (botCodeDetails is null)
             {
-                var botId = rgBot.id;
-                var localBot = RGBotAssetsManager.GetInstance()?.GetBotAssetRecord(botId);
-                var md5 = ZipLocalBot(botId, localBot.Path);
-                localMD5[botId] = md5;
+                // Unlikely unless the server is having a problem, but we should handle it.
+                // Just don't sync this bot
+                RGDebug.LogWarning($"Server reported no code for bot {remoteBot.id}. Skipping sync.");
+                return false;
             }
 
-            return localMD5;
-        }
-
-        private async Task<bool> UpdateBotZipOnRG(long botId, string md5)
-        {
-            var didUpdate = false;
-            // if zip checksums don't match.. push new zip files
-            // Get remote zip details, check md5sum, decide if need to push
-            var needsToUpload = false;
+            // If the checksums match, we're done
+            if (localMd5 == botCodeDetails.md5)
+            {
+                RGDebug.LogDebug(
+                    $"RG Bot Id: {remoteBot.id} is up to date; local md5: {localMd5} == remote md5: {botCodeDetails.md5}");
+                return false;
+            }
 
-            await _rgServiceManager.GetBotCodeDetails(
-                botId,
-                (botCodeDetails) =>
-                {
-                    if (md5 != botCodeDetails.md5)
-                    {
-                        RGDebug.LogDebug(
-                            $"RG Bot Id: {botId} needs to update remote zip; local md5: {md5} != remote md5: {botCodeDetails.md5}");
-                        needsToUpload = true;
-                    }
-                },
-                () =>
-                {
-                    // not found.. need to push for first time
-                    needsToUpload = true;
-                }
-            )!;
+            var localTime = localLastUpdated is null
+                ? "<<unknown>>" // Very unlikely, indicates the local files don't exist.
+                : localLastUpdated.Value.ToLocalTime().ToString("ddd MMM d, yyyy HH:mm:ss tt");
+            var remoteTime = botCodeDetails.modifiedDate is null
+                ? "<<unknown>>" // Very unlikely, indicates the server is misbehaving.
+                : botCodeDetails.modifiedDate.Value.ToLocalTime().ToString("ddd MMM d, yyyy HH:mm:ss tt");
```

Comment:
This is definitely a case where I'd consider throwing, but since we're in the user's Unity editor I hesitated. If we had some telemetry, I'd have it log an "error"-level message but then recover by showing `<<unknown>>`.

Reply:
to answer @vontell 's question.  BotCodeDetails is the metadata about the zip file itself.  Yes, it gets updated on any zip change on the server.